src/components/Settings/Language/Language.container.js

```diff
@@ -418,86 +453,16 @@ export class LanguageContainer extends Component {
           ttsError: true
         }
       });
-      return;
-    }
-    if (ttsEngine.name !== engineName) {
-      try {
-        await this.handleSetTtsEngine(engineName);
-      } catch {
-        this.setState({
-          downloadingLangError: {
-            ttsError: false,
-            langError: true
-          }
-        });
-      }
-    }
-    const localLangs = this.props.localLangs;
-    if (!localLangs.includes(selectedLang)) {
+    } else if (
```

Comment:
return if (!ttsEnginesNames.includes(engineName)) 
and remove else if condition. if any of before conditions would be true only set:
```
        downloadingLangError: {
          ttsError: false,
          langError: true
        }
```

Reply:
can you rephrase your comment? I don't understand.

Reply:
Sorry. If ttsEnginesNames does not include the engine name we could return from the function after set ttsError to true. if not, We know that is a langError and the else if condition would be ambiguous.